packages/jest-haste-map/src/ModuleMap.ts

```diff
@@ -36,6 +35,30 @@ export default class ModuleMap {
   private readonly _raw: RawModuleMap;
   private json: SerializableModuleMap | undefined;
 
+  private static mapToArrayRecursive(
+    map: Map<any, any>,
+  ): Array<[string, unknown]> {
+    let arr = Array.from(map);
+    if (arr[0] && arr[0][1] instanceof Map) {
+      arr = arr.map(
+        el => [el[0], this.mapToArrayRecursive(el[1])] as [string, unknown],
+      );
+    }
+    return arr;
+  }
+
+  private static mapFromArrayRecursive(
+    arr: ReadonlyArray<[string, unknown]>,
+  ): Map<string, unknown> {
+    if (arr[0] && Array.isArray(arr[1])) {
```

Comment:
Why do you check for `arr[1]`? Is there some constraint that there are always at least 2 items in arr when going recursive?
Checking for `arr[0][1]` would make more sense.